coop_memberspace/controllers/main.py

```diff
@@ -0,0 +1,451 @@
+                "user": user,
+            },
+        )
+
+    @http.route("/myteam", type="http", auth="user", website=True)
+    def page_myteam(self, **kwargs):
+        user = request.env.user
+        tmpl_lines = user.partner_id.tmpl_reg_line_ids.filtered(
+            lambda r: r.is_current
+        )
+        shift_tmpl = tmpl_lines and tmpl_lines[0].shift_template_id or False
+        coordinators = (
+            shift_tmpl
+            and shift_tmpl.user_ids.sorted(key=lambda r: r.name)
+            or []
+        )
+        members = (
+            shift_tmpl
+            and shift_tmpl.registration_ids.filtered(
+                lambda r: r.is_current_participant
+            )
+            .mapped("partner_id")
+            .filtered(lambda r: r.shift_type == "standard")
+            .sorted(key=lambda r: r.name)
+            or []
+        )
+
+        alias_leader = (
+            shift_tmpl
+            and request.env["memberspace.alias"].search(
+                [
+                    ("shift_id", "=", shift_tmpl.id),
+                    ("type", "=", "coordinator"),
+                ],
+                limit=1,
+            )
+            or False
+        )
+        alias_leader = (
+            alias_leader and alias_leader[0].alias_id.name_get()[0][1] or ""
+        )
+
+        alias_team = (
+            shift_tmpl
+            and request.env["memberspace.alias"].search(
+                [("shift_id", "=", shift_tmpl.id), ("type", "=", "team")],
+                limit=1,
+            )
+            or False
+        )
+        alias_team = (
+            alias_team and alias_team[0].alias_id.name_get()[0][1] or ""
+        )
+
+        is_leader = user.partner_id in coordinators
+
+        return request.render(
+            "coop_memberspace.myteam",
+            {
+                "coordinators": coordinators,
+                "members": members,
+                "alias_team": alias_team,
+                "alias_leader": alias_leader,
+                "is_leader": is_leader,
+                "current_member": user.partner_id,
+            },
+        )
+
+    @http.route("/profile", type="http", auth="user", website=True)
+    def page_myprofile(self, **kwargs):
+
+        return request.render(
+            "coop_memberspace.myprofile", {"user": request.env.user}
+        )
+
+    @http.route("/statistics", type="http", auth="user", website=True)
+    def page_statistics(self, **kwargs):
+        first_day_of_year = datetime.now().strftime("%Y-01-01 00:00:00")
+        end_day_of_year = datetime.now().strftime("%Y-12-31 23:59:59")
+        sales = (
+            request.env["pos.order"]
+            .sudo()
+            .search(
+                [
+                    ("state", "in", ["paid", "done", "invoiced"]),
+                    ("date_order", ">=", first_day_of_year),
+                    ("date_order", "<=", end_day_of_year),
+                ]
+            )
+        )
```

Comment:
If I'm not mistaken this `sales` recordset is only used to get the number of records: `len(sales)`
You could replace this `search()` by a `search_count()` instead, to improve performance